crates/router/src/connector/shift4/transformers.rs

```diff
@@ -671,6 +671,7 @@ impl<F>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
+        let token_id = Some(item.response.token.id.clone());
```

Comment:
This token id is not connector response id , it is shift4 3ds token id . revert the changes from shift4 try_from .